django/contrib/admin/options.py

```diff
@@ -473,9 +473,12 @@ def to_field_allowed(self, request, to_field):
             for inline in admin.inlines:
                 registered_models.add(inline.model)
 
-        for related_object in (opts.get_all_related_objects(include_hidden=True) +
-                               opts.get_all_related_many_to_many_objects()):
-            related_model = related_object.model
+        related_objects = (
+            f for f in opts.get_fields(include_hidden=True)
+            if (f.auto_created and not f.concrete)
```

Comment:
@PirosB3 There's one significant difference - a `GenericRelation` is backed by a `GenericForeignKey` definition on the other model; a theoretical `ManyToOneField` doesn't have anything concrete on the other end to provide the "actual" definition. `GenericRelation` objects can be ignored for anything other than data lookup purposes - anything to do with deletion etc can be ignored. `ManyToOneField` would need to invoke some functionality.

So - same argument as above - do we use what is there right now; try to fix it for an example that doesn't exist, or use a more explicit syntax that will be known to be incorrect in a theoretical case, but not for any actually extant case?